homeassistant/components/google/config_flow.py

```diff
@@ -0,0 +1,135 @@
+"""Config flow for Google integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from oauth2client.client import Credentials
+
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .api import DeviceFlow, OAuthError, async_create_device_flow
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle Google Calendars OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+
+    def __init__(self) -> None:
+        """Set up instance."""
+        super().__init__()
+        self._reauth = False
+        self._device_flow: DeviceFlow | None = None
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    async def async_step_import(self, info: dict[str, Any]) -> FlowResult:
+        """Import existing auth from Nest."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        implementations = await config_entry_oauth2_flow.async_get_implementations(
+            self.hass, self.DOMAIN
+        )
+        assert len(implementations) == 1
+        self.flow_impl = list(implementations.values())[0]
+        self.external_data = info
+        return await super().async_step_creation(info)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle external yaml configuration."""
+        if not self._reauth and self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        await self.async_set_unique_id(DOMAIN)
```

Comment:
Why do we set a unique id if we only allow a single entry? We could just guard against more than one entry as we do in the import step.

Reply:
My assumption was this was required, so no particular reason. Removing.